client/packages/common/src/ui/components/inputs/TextInput/NumericTextInput.test.tsx

```diff
@@ -91,4 +91,48 @@ describe('Test NumericTextInput component', () => {
     fireEvent.blur(input);
     expect(input).toHaveValue('1.100');
   });
+
+  it('should handle negative number input', async () => {
+    const { getByRole } = render(<TestNumericTextInput allowNegative />);
+    const input = getByRole('textbox');
+
+    fireEvent.change(input, { target: { value: '-' } });
+    expect(input).toHaveValue('-');
+    fireEvent.change(input, { target: { value: '-5' } });
+    expect(input).toHaveValue('-5');
+    fireEvent.blur(input);
+    expect(input).toHaveValue('-5');
+  });
+
+  it('should handle removing input', async () => {
+    const { getByRole } = render(<TestNumericTextInput />);
+    const input = getByRole('textbox');
+
+    fireEvent.change(input, { target: { value: '500' } });
+    expect(input).toHaveValue('500');
+    fireEvent.change(input, { target: { value: '' } });
+    expect(input).toHaveValue('');
+    fireEvent.blur(input);
+    expect(input).toHaveValue('');
+  });
+
+  it('should format large numbers', async () => {
+    const { getByRole } = render(<TestNumericTextInput />);
+    const input = getByRole('textbox');
+
+    fireEvent.change(input, { target: { value: '1000' } });
+    expect(input).toHaveValue('1,000');
+    fireEvent.blur(input);
+    expect(input).toHaveValue('1,000');
+  });
+
+  it('should not format large numbers when explicitly prevented', async () => {
+    const { getByRole } = render(<TestNumericTextInput noFormatting />);
+    const input = getByRole('textbox');
+
+    fireEvent.change(input, { target: { value: '1000' } });
+    expect(input).toHaveValue('1000');
+    fireEvent.blur(input);
+    expect(input).toHaveValue('1000');
+  });
```

Comment:
Confirms formatting prop behaves as expected

Reply:
Not introduced in this PR, but noticed that there is a warning with this test (`A component is changing an uncontrolled input to be controlled`)

The `noFormatting` is only used in printer label settings, where a value is required, so we don't currently have a use case where we would get this warning in the app... maybe can just capture in carry over somewhere?

That printer label settings is having the same issue (can't delete the last digit) but I think it's due to the onChange implementation rather than the input... do you want to have a look now or I can create a separate issue?